### PR TITLE
set up associations

### PR DESCRIPTION
**Pull request by AbdelazizSoliman**
In this pull request I:

- [ ] Created class classroom
- [ ] Updated Classroom and Student with a has-many/belongs-to relationship
- [ ] Created class book
- [ ] Created class rental
- [ ] Created person-book many-to-many relationships via rental class

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
 require_relative 'person'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(classroom, age, name = 'unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
```
Comment (julie-ify) [inline student.rb:10]: - [ ] The association between the student and classroom must also be its own method. Kindly move this to a method.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, date)
+    rental = Rental.new(date, self, person)
+    @rentals << rental
+    person.add_rental(rental)
+  end
```
Comment (julie-ify) [inline book.rb]: - [ ] You have a lot of unnecessary code here. All we need is the first instantiation on line 11. The rest are repetitive of the first line. Creating a rental on line 11, already adds it to the rentals array in the Rental class, so there's no need for line 12. There is also no need to call `add_rental` in line 13 (you are not calling it correctly either), the methods should be independent of each other.

  ```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @AbdelazizSoliman,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 3]
```diff
@@ -1,9 +1,16 @@
 require_relative 'person'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(classroom, age, name = 'unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    add_to_classroom
```
Comment (1GORDON) [inline student.rb]: Great job @AbdelazizSoliman ✋

- [ ] You have done great implementing the requested changes. However, you do not need to call this `add-to-classroom` method in the constructor. Could you maintain this method separate from the constructor?
add-to-classroom

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @AbdelazizSoliman ✋

Good job so far!

## Highlights!
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @AbdelazizSoliman, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to refactor the code as requested you previous reviewers.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**